scripts/build_shapes.py

```diff
@@ -0,0 +1,312 @@
+        else:
+            polys = mainpoly
+    return polys.simplify(tolerance=tolerance)
+
+
+def countries(update=False):
+    countries = snakemake.config["countries"]
+
+    # download data if needed and get the layer id 0, corresponding to the countries
+    df_countries = get_GADM_layer(countries, 0, update)
+
+    # select and rename columns
+    df_countries = df_countries[["GID_0", "geometry"]].copy()
+    df_countries.rename(columns={"GID_0": "name"}, inplace=True)
+
+    # set index and simplify polygons
+    ret_df = df_countries.set_index("name")["geometry"].map(_simplify_polys)
+
+    return ret_df
+
+
+def country_cover(country_shapes, eez_shapes=None):
+    shapes = list(country_shapes)
+    if eez_shapes is not None:
+        shapes += list(eez_shapes)
+
+    europe_shape = cascaded_union(shapes)
+    if isinstance(europe_shape, MultiPolygon):
+        europe_shape = max(europe_shape, key=attrgetter("area"))
+    return Polygon(shell=europe_shape.exterior)
+
+
+def save_to_geojson(df, fn):
+    if os.path.exists(fn):
+        os.unlink(fn)  # remove file if it exists
+    if not isinstance(df, gpd.GeoDataFrame):
+        df = gpd.GeoDataFrame(dict(geometry=df))
+
+    # save file if the GeoDataFrame is non-empty
+    if df.shape[0] > 0:
+        df = df.reset_index()
+        schema = {**gpd.io.file.infer_schema(df), "geometry": "Unknown"}
+        df.to_file(fn, driver="GeoJSON", schema=schema)
+
+
+def load_EEZ(selected_countries_codes, name_file="eez_v11.gpkg"):
+    EEZ_gpkg = os.path.join(os.path.dirname(os.getcwd()), "data", "raw", "eez",
+                            name_file)  # Input filepath gpkg
+
+    if not os.path.exists(EEZ_gpkg):
+        raise Exception(
+            f"File EEZ {name_file} not found, please download it from https://www.marineregions.org/download_file.php?name=World_EEZ_v11_20191118_gpkg.zip and copy it in {os.path.dirname(EEZ_gpkg)}"
+        )
+
+    geodf_EEZ = gpd.read_file(EEZ_gpkg)
+    geodf_EEZ.dropna(axis=0, how="any", subset=["ISO_TER1"], inplace=True)
+    # [["ISO_TER1", "TERRITORY1", "ISO_SOV1", "ISO_SOV2", "ISO_SOV3", "geometry"]]
+    geodf_EEZ = geodf_EEZ[["ISO_TER1", "geometry"]]
+    geodf_EEZ = geodf_EEZ[[
+        any([_three_2_two_digits_country(x) in selected_countries_codes])
+        for x in geodf_EEZ["ISO_TER1"]
+    ]]
+    geodf_EEZ.reset_index(drop=True, inplace=True)
+
+    geodf_EEZ.rename(columns={"ISO_TER1": "name"}, inplace=True)
+
+    return geodf_EEZ
+
+
+def eez(update=False, tol=1e-3):
+    countries = snakemake.config["countries"]
+
+    # load data
+    df_eez = load_EEZ(countries)
+
+    # set index and simplify polygons
+    ret_df = df_eez.set_index("name")["geometry"].map(
+        lambda s: _simplify_polys(s, filterremote=False))
+    ret_df = gpd.GeoSeries({
+        k: v
+        for k, v in ret_df.iteritems() if v.distance(country_shapes[k]) < tol
+    })
+    ret_df.index.name = "name"
+
+    return ret_df
+
+
+def nuts3(update=False):
```

Comment:
Would remove nuts3 and any EEZ (European Economic Zone) naming, wherever it makes sense at the moment.
These are EU namings. I would name it "gadm" and "aez". 

As you said, we need to work on the GADM level description. Guess that can be a function input somewhere. But that's work ongoing. Otherwise, it looks good for now :+1: Well done!

Reply:
Ok! I agree, I kept the original nomenclature on the similarities with PyPSA-Eur.
Shall we define "Economic Zones" in general?
Regarding gadm, also countries are derived from GADM, but maybe we can name them gadm to refer to the small regions.
I will proceed as discussed